### ISSUE TITLE
Add a resource for firewalld

### ISSUE THREAD

**Issue by tas50** [chef/chef#12269]
We have a pretty clear path to legacy firewall management with the iptables resources, but newer systems use the firewalld project so we need to make sure we create a resource for firewall management on these newer systems.

- CentOS 7 and newer
- Fedora 18 and newer
- OpenSUSE Leap 15 and newer
- Red Hat Enterprise Linux 7 and newer
- SUSE Linux Enterprise 15 and newer

**Comment by tas50** [comment]
https://github.com/sjsadowski/firewalld-cookbook

**Comment by ramereth** [comment]
FWIW I think https://github.com/sous-chefs/firewall also has some code for firewalld.

**Comment by tas50** [comment]
The abstraction of the firewall cookbook isn't something we want to move into Chef though. Firewalls are different and we should just embrace that. In the past when we've tried magical abstraction it's ended in sadness for everyone.

**Comment by ramereth** [comment]
@tas50 yeah, I wouldn't recommend you do that either, but I wasn't sure if there was any code in there that might be useful for this too specifically on firewalld. I'm not suggesting you import that whole cookbook :)

**Comment by tas50** [comment]
I'll take a look to see if there's anything we can lift :)
